src/sanitizer.js

```diff
@@ -320,6 +339,12 @@ export function isValidAttr(tagName, attrName, attrValue) {
     return false;
   }
 
+  if (attrName == 'class' &&
+      attrValue &&
+      /(^|\W)i-amphtml-/i.test(attrValue)) {
+    return false;
+  }
```

Comment:
Looks like Caja doesn't blacklist `class` in the first place. Tried this on https://validator.ampproject.org/ and it works.